app/lib/lets-encrypt.server.ts

```diff
@@ -29,8 +84,112 @@ class LetsEncrypt {
       directoryUrl: this.#directoryUrl,
       accountKey: this.#accountKey,
     });
+
+    await this.#registerReloadAccount();
+
+    return this;
   };
 
+  /**
+   * This fn extracts and requests challenge information from / based on
+   * the authorizations array that we got in the order.
+   */
+  #extractChallenges = async () => {
+    this.#authorizations = await this.#client!.getAuthorizations(this.#order!);
+
+    /**
+     * Each authorization (a soon to be common name or alt name
+     * we requested in createOrder => identifiers) contains one or
+     * more challenges. At least one challenge per authorization needs
+     * to be passed for the authorization to be granted
+     *
+     * Later, we can use the granted authorizations in our certificate
+     */
+    const promises = this.#authorizations.map(async (authorization) => {
+      const selectedChallenge = authorization.challenges.find(
+        ({ type }) => type === 'dns-01'
+      ) as AcmeDnsChallenge;
+
+      if (!selectedChallenge)
```

Comment:
For the future, we have a module in the Blue Stack to help with this: [tiny-invariant](https://github.com/alexreardon/tiny-invariant).  You'd use it like this:

```ts
import invariant from 'tiny-invariant';
...
invariant(selectedChallenge, 'The authorization object does not contain dns-01 type challenge. This should never happen.');
```

It's a nice utility to have, since this pattern happens all the time with TS.  After your call to `invariant(foo)` you can use `foo` without worry.

Reply:
I have used invariant before, but does it also satisfy the typescript built in null check? I.e.,
```
if (!a) return;
  // I can use a after this without specifying a!
```
Does this hold true for invariant too?

Reply:
Yes, it's perfect for this.